### ISSUE TITLE
RUF029 is throwing false positives on async iterators since v0.4.2

### ISSUE THREAD

**Issue by frgfm** [astral-sh/ruff#11218]
Hello there :wave: 

Kudos for this great tool, it's more than useful for so many projects.
I noticed that with the latest release, ruff is throwing errors on RUF029 on async interators.
The snippet below:
```python
import asyncio
from typing import Tuple


async def example(time: int) -> Tuple[int, int, int]:
    await asyncio.sleep(time)
    return (1, 2, 3)


async def my_function() -> str:
    vals = [str(val) for val in await example(1)]
    return ",".join(vals)
```
is working alright with v0.4.1. But when running v0.4.2, it raises:
```
RUF029 Function `my_function` is declared `async`, but doesn't `await` or use `async` features.
```

Now the strange part is that I looked around and found #11070 which supposedly fixes this exact problem mentioned in #11043. I've tried to downgrade and upgrade several times, I'm getting the same problem consistently on 0.4.2 but not on 0.4.1.

Am I misunderstanding #11043 and this isn't the same problem? Or did the fix not cover everything?

I'm not proficient with Rust, but happy to help if I can!

Cheers!

**Comment by charliermarsh** [comment]
Apologies, I see the issue -- will fix now.

**Comment by zanieb** [comment]
Fixed in https://github.com/astral-sh/ruff/pull/11219, I presume?

**Comment by frgfm** [comment]
Don't be sorry @charliermarsh, you're doing an amazing job with ruff! And it's all the more commendable to address & fix reported issues that fast :clap:

Thanks for the fix!
